File: src/csharp/Foundgine.Core/Semantic/Mutation/SemanticMutationValueReference.cs
using Foundgine.Core.Abstractions;

namespace Foundgine.Core.Semantic.Mutation;

/// <summary>
/// References a semantic value produced by an earlier mutation operation.
/// This describes dependency intent, not provider execution mechanics.
/// </summary>
public sealed record SemanticMutationValueReference(
    int SourceOperationIndex,
    FieldId SourceField);

File: src/java/foundgine-core/src/main/java/com/foundgine/core/semantic/mutation/SemanticMutationValueReference.java
package com.foundgine.core.semantic.mutation;

import com.foundgine.core.abstractions.FieldId;

/** References a semantic value produced by an earlier mutation operation. */
public record SemanticMutationValueReference(int sourceOperationIndex, FieldId sourceField) {
	public SemanticMutationValueReference {
		if (sourceOperationIndex < 0)
			throw new IllegalArgumentException("sourceOperationIndex must be non-negative.");
	}
}
